kv/kvserver: TestReplicateQueueExpirationLeasesOnly failed

kv/kvserver.TestReplicateQueueExpirationLeasesOnly [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9122901?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9122901?buildTab=artifacts#/) on master @ [0a72a49eb6e4ce5823b53d02125441c3da2e94f5](https://github.com/cockroachdb/cockroach/commits/0a72a49eb6e4ce5823b53d02125441c3da2e94f5):


```
=== RUN   TestReplicateQueueExpirationLeasesOnly
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueExpirationLeasesOnly2320733262
    test_log_scope.go:79: use -show-logs to present logs inline
    replicate_queue_test.go:2406: initial: epochLeases=316 expLeases=2
    replicate_queue_test.go:2411: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/replicate_queue_test.go:2411
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueExpirationLeasesOnly
    panic.go:522: -- test log scope end --
--- FAIL: TestReplicateQueueExpirationLeasesOnly (319.95s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueExpirationLeasesOnly.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25608

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -2361,6 +2361,11 @@ func TestReplicateQueueExpirationLeasesOnly(t *testing.T) {
 	skip.UnderRace(t) // too slow under stressrace
 	skip.UnderShort(t)
 
+	timeout := 5 * time.Second
+	if skip.Stress() {
+		timeout = 30 * time.Second
+	}
+
 	ctx := context.Background()
 	tc := testcluster.StartTestCluster(t, 3, base.TestClusterArgs{
 		ServerArgs: base.TestServerArgs{
@@ -2410,12 +2415,12 @@ func TestReplicateQueueExpirationLeasesOnly(t *testing.T) {
 	require.NoError(t, err)
 	require.Eventually(t, func() bool {
 		epochLeases, expLeases = countLeases()
+		t.Logf("enabling: epochLeases=%d expLeases=%d", epochLeases, expLeases)
 		return epochLeases == 0 && expLeases > 0
-	}, 10*time.Second, 200*time.Millisecond)
-	t.Logf("enabled: epochLeases=%d expLeases=%d", epochLeases, expLeases)
+	}, timeout, 500*time.Millisecond)
 
 	// Run a scan across the ranges, just to make sure they work.
-	scanCtx, cancel := context.WithTimeout(ctx, 3*time.Second)
+	scanCtx, cancel := context.WithTimeout(ctx, timeout)
 	defer cancel()
 	_, err = db.Scan(scanCtx, scratchKey, scratchKey.PrefixEnd(), 1)
 	require.NoError(t, err)
@@ -2429,7 +2434,7 @@ func TestReplicateQueueExpirationLeasesOnly(t *testing.T) {
 	require.NoError(t, err)
 	require.Eventually(t, func() bool {
 		epochLeases, expLeases = countLeases()
+		t.Logf("disabling: epochLeases=%d expLeases=%d", epochLeases, expLeases)
 		return epochLeases > 0 && expLeases > 0 && expLeases <= initialExpLeases
-	}, 10*time.Second, 200*time.Millisecond)
-	t.Logf("disabled: epochLeases=%d expLeases=%d", epochLeases, expLeases)
+	}, timeout, 500*time.Millisecond)
 }

